eng/regenerate-third-party-notices.proj

```diff
@@ -29,7 +30,6 @@
           dotnet/icu
```

Comment:
rather than hard code which don't have one, can the tool simply scan them all, and include what it finds? that way we don't have to remember to edit this later?

Reply:
I've created the issue to update the related tooling: https://github.com/dotnet/runtime/issues/74766